grails-app/controllers/org/pih/warehouse/shipping/ShipmentController.groovy

```diff
@@ -888,13 +888,13 @@ class ShipmentController {
 
 
     def saveComment() {
-        def shipmentInstance = Shipment.get(params.shipmentId)
-        def recipient = (params.recipientId) ? User.get(params.recipientId) : null
-        def comment = new Comment(comment: params.comment, sender: session.user, recipient: recipient)
-        if (shipmentInstance) {
-            shipmentInstance.addToComments(comment).save(flush: true)
-            flash.message = "${warehouse.message(code: 'shipping.addedCommentToShipment.message', args: [params.comment, shipmentInstance.name])}"
+        Shipment shipmentInstance = Shipment.get(params.shipmentId)
+        if (!shipmentInstance) {
+            throw new Exception("Could not locate shipment with ID " + params.shipmentId)
```

Comment:
maybe let's come up with a more specific exception? using the general `Exception` doesn't feel right to me.
Maybe we should start creating our own exceptions and define them? e.g. `NotFoundException`?

Reply:
I believe `org.hibernate.ObjectNotFoundException` could be a good candidate

Reply:
:+1: